File: menu_backend/menu/Models/PlaceModels/Place.cs
using where.Models.CategoryModels;

namespace where.Models.PlaceModels
{
    public class Place
    {
        public string id { get; set; }
        public string name { get; set; }
        public string location { get; set; }
        public List<string>menuImages { get; set; }
        public Category category { get; set; }
        public Dictionary<string, int> userRates { get; set; }
        public Place(string id, string name, string location, List<string> menuImages, Category category, Dictionary<string, int> userRates)
        {
            this.id = id;
            this.name = name;
            this.location = location;
            this.menuImages = menuImages;
            this.category = category;
            this.userRates = userRates;
        }
    }
}

File: menu_mobile/lib/features/Home/model/entities/Place.dart
import 'package:where/features/Home/model/entities/Category.dart';
import 'dart:convert';

class Place {
  String id;
  String name;
  String location;
  List<String> menuImages;
  Category category;
  Map<String, int> userRates;

  Place(
      {required this.id,
      required this.name,
      required this.location,
      required this.menuImages,
      required this.category,
      required this.userRates});

  factory Place.fromJson(Map<String, dynamic> jsondata) {
    List<String> menuImagesdata = [];
    jsondata["menuImages"].forEach((element) {
      menuImagesdata.add(element.toString());
    });
    Map<String, int> userRatesdata = {};
    jsondata["userRates"].forEach((key, value) {
      userRatesdata[key] = int.parse(value.toString());
    });
    return Place(
        id: jsondata["id"] ?? '',
        name: jsondata["name"] ?? '',
        location: jsondata["location"],
        menuImages: menuImagesdata,
        category: Category.fromJson(jsondata["category"]),
        userRates: userRatesdata);
  }

  Map<String, dynamic> toMap() => {
        "id": id,
        "name": name,
        "location": location,
        "menuImages": menuImages,
        "category": category,
        "userRates": userRates
      };
}
